src/sonic-yang-models/yang-models/sonic-pbh.yang

```diff
@@ -0,0 +1,218 @@
+module sonic-pbh {
+
+	yang-version 1.1;
+
+	namespace "http://github.com/Azure/sonic-pbh";
+	prefix pbh;
+
+	import ietf-inet-types {
+		prefix inet;
+	}
+
+	import sonic-port {
+		prefix port;
+	}
+
+	import sonic-portchannel {
+		prefix lag;
+	}
+
+	description "PBH YANG Module for SONiC OS";
+
+	revision 2021-04-23 {
+		description "First Revision";
+	}
+
+	container sonic-pbh {
+
+		container PBH_HASH {
+
+			description "PBH_HASH part of config_db.json";
+
+			list PBH_HASH_LIST {
+
+				key "hash_name";
+
+				leaf hash_name {
+					description "The name of this hash";
+					type string;
+				}
+
+				leaf hash_field {
+					description "Configures native hash filed for this hash";
+					mandatory true;
+					type string {
+						pattern "INNER_IP_PROTOCOL|INNER_L4_DST_PORT|INNER_L4_SRC_PORT|INNER_DST_IPV4|INNER_SRC_IPV4|INNER_DST_IPV6|INNER_SRC_IPV6";
+					}
+				}
+
+				leaf ipv4_mask {
+					description "Configures IPv4 address mask for this hash";
+					mandatory true;
+					type inet:ipv4-address;
+					when "boolean(../hash_field[.='INNER_DST_IPV4' or .='INNER_SRC_IPV4'])";
+				}
+
+				leaf ipv6_mask {
+					description "Configures IPv6 address mask for this hash";
+					mandatory true;
+					type inet:ipv6-address;
+					when "boolean(../hash_field[.='INNER_DST_IPV6' or .='INNER_SRC_IPV6'])";
+				}
+
+				leaf sequence_id {
+					description "Configures in which order the fields are hashed and defines which fields should be associative";
+					mandatory true;
+					type uint32;
+				}
+
+			}
+			/* end of PBH_HASH_LIST */
+		}
+		/* end of container PBH_HASH */
+
+		container PBH_RULE {
+
+			description "PBH_RULE part of config_db.json";
+
+			list PBH_RULE_LIST {
+
+				key "table_name rule_name";
+
+				leaf table_name {
+					description "The name of table which holds this rule";
+					type leafref {
+						path "/pbh:sonic-pbh/pbh:PBH_TABLE/pbh:PBH_TABLE_LIST/pbh:table_name";
+					}
+				}
+
+				leaf rule_name {
+					description "The name of this rule";
+					type string;
+				}
+
+				leaf priority {
+					description "Configures priority for this rule";
+					mandatory true;
+					type uint32;
+				}
+
+				leaf gre_key {
+					description "Configures packet match for this rule: GRE key";
+					type string {
+						pattern "(0x){1}[a-fA-F0-9]+/(0x){1}[a-fA-F0-9]+";
```

Comment:
Don't we have to escape the "/" using "\", in this regex pattern to accept an input like this (0xa124/0x1b)?
Asking something like this with "\" escape char in "pattern "(0x){1}[a-fA-F0-9]+\/(0x){1}[a-fA-F0-9]+"";

Reply:
@rathnasabapathyv No, we don't. This was tested with DPB. No issues observed.